CMakeLists.txt

```diff
@@ -68,6 +68,14 @@ if(MPI_LINK_FLAGS)
     set(CMAKE_LINK_FLAGS ${MPI_LINK_FLAGS} ${CMAKE_LINK_FLAGS})
 endif()
 
+#### FORCE MPI Libraries to front by appending to CMAKE_EXE_LINKER_FLAGS
+if(MPI_CXX_LIBRARIES)
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${MPI_CXX_LINK_FLAGS}")
+    foreach(ITEM ${MPI_CXX_LIBRARIES})
+        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ITEM}")
+    endforeach()
+endif()
```

Comment:
Here's the relevant file in CMake... I had to integrate to rearrange the mpi linker flags.